test/core/stress.test.ts

```diff
@@ -36,4 +36,21 @@ describe('stress tests', () => {
     });
     expect(result).toBeDefined();
   });
+  test('countPathsToRoot() is cached', async () => {
+    const graph = await generateLargeGraph(100_000);
+    let start = Date.now();
+    graph.countPathsToRoot({
+      name: dependencyName,
+      version: '1.2.3',
+    });
+    const firstCallDuration = Date.now() - start;
+    start = Date.now();
+    graph.countPathsToRoot({
+      name: dependencyName,
+      version: '1.2.3',
+    });
+    const secondCallDuration = Date.now() - start;
+    // the second call must be *significantly* faster than the first
+    expect(secondCallDuration).toBeLessThan(firstCallDuration * 0.2);
```

Comment:
Couldn't this be a flaky test waiting to happen? If you want to test for cache, maybe a spy on the cache would produce a more reproducible test? 

Just a thought.

Reply:
Given JS is single threaded, and these calls are all synchronous, I think we'll be ok